components/ILIAS/OrgUnit/classes/Settings/class.ilOrgUnitObjectPositionSetting.php

```diff
@@ -16,93 +17,109 @@
  ********************************************************************
  */
 
-/* Copyright (c) 1998-2010 ILIAS open source, Extended GPL, see docs/LICENSE */
+declare(strict_types=1);
 
 /**
- * Stores object activation status of orgunit position settings.
- * @author Stefan Meyer <smeyer.ilias@gmx.de>
+ * Object-specific activation settings for orgunit position.
  */
 class ilOrgUnitObjectPositionSetting
 {
     protected ilDBInterface $db;
-    private int $obj_id;
     private ?bool $active = null;
+    private ?ilOrgUnitObjectTypePositionSetting $settings_cache = null;
 
-    public function __construct(int $a_obj_id)
-    {
+    public function __construct(
+        private int $obj_id
+    ) {
         $this->db = $GLOBALS['DIC']->database();
-        $this->obj_id = $a_obj_id;
         $this->readSettings();
     }
 
-    /**
-     * Lookup activation status
-     */
-    public function lookupActive(int $a_obj_id): bool
+    public static function getFor(int $obj_id): self
     {
-        $db = $GLOBALS['DIC']->database();
+        return new self($obj_id);
+    }
+
+    public function isActive(): bool
+    {
+        if (! $this->isGloballyEnabled()) {
+            return false;
+        }
 
-        $query = 'select *  from orgu_obj_pos_settings ' . 'where obj_id = '
-            . $db->quote($a_obj_id, 'integer');
-        $res = $this->db->query($query);
-        while ($row = $res->fetchRow(ilDBConstants::FETCHMODE_OBJECT)) {
-            return (bool) $row->active;
+        if (! $this->isChangeable()) { //implicit: && isGloballyEnabled == true
+            return true;
         }
+
+        if ($this->active === null) { //implicit: isGloballyEnabled && isChangeable
+            return (bool) $this->getGlobalSettingsForThisObjectType()->getActivationDefault();
+        }
+
+        return $this->active;
     }
 
     /**
-     * Check if position access is active. This returns true or false if it is object specific or null if the object has no setting.
+     * convenience wrapper for specific objects to check on global settings
      */
-    public function isActive(): ?bool
+    public function isGloballyEnabled(): bool
```

Comment:
This naming throw me off when reading `isActive`. I thought this was about whether the org units are activated at all globally... So please rename to `isActiveForOwnType`.